src/Umbraco.Infrastructure/Services/Implement/ContentService.cs

```diff
@@ -267,8 +255,20 @@ public IContent CreateAndSave(string name, int parentId, string contentTypeAlias
                 if (parentId > 0 && parent == null)
                     throw new ArgumentException("No content with that id.", nameof(parentId)); // causes rollback
 
-                var content = parentId > 0 ? new Content(name, parent, contentType) : new Content(name, parentId, contentType);
-                CreateContent(scope, content, userId, true);
+                var content = parentId > 0 ? new Content(name, parent, contentType, userId) : new Content(name, parentId, contentType, userId);
+                var evtMsgs = EventMessagesFactory.Get();
+
+                // if saving is cancelled, content remains without an identity
+                var saveEventArgs = new ContentSavingEventArgs(content, evtMsgs);
+                if (!scope.Events.DispatchCancelable(Saving, this, saveEventArgs, nameof(Saving)))
+                {
+                    _documentRepository.Save(content);
+
+                    scope.Events.Dispatch(Saved, this, saveEventArgs.ToContentSavedEventArgs(), nameof(Saved));
+                    scope.Events.Dispatch(TreeChanged, this, new TreeChange<IContent>(content, TreeChangeTypes.RefreshNode).ToEventArgs());
+
+                    Audit(AuditType.New, content.CreatorId, content.Id, $"Content '{content.Name}' was created with Id {content.Id}");
+                }
```

Comment:
I prefer we reuse the `Save` method, even that the `AuditType` will be `AuditType.Save` instead of new.